nucleus/payara-modules/payara-micro/src/main/java/fish/payara/micro/PayaraMicro.java

```diff
@@ -592,7 +592,7 @@ private void scanArgs(String[] args) {
                 }
                 i++;
             } else if ("--domainConfig".equals(arg)) {
-                alternateDomainXML = new File(args[i] + 1);
+                alternateDomainXML = new File(args[i + 1]);
                 if (!alternateDomainXML.exists() || !alternateDomainXML.isFile() || !alternateDomainXML.canRead() || !alternateDomainXML.getAbsolutePath().endsWith(".xml")) {
                     System.err.println(alternateDomainXML.getAbsolutePath() + " is not a valid path to an xml file and will be ignored");
```

Comment:
I'd go for fail-fast behaviour here. If you explicitly ask for a non-existing file then it should fail to start.